app/controllers/provider_interface/offer/courses_controller.rb

```diff
@@ -45,6 +47,12 @@ def update
 
     private
 
+      def reset_study_mode_if_course_has_changed_and_update_course
+        new_val = course_params['course_id']
+        @wizard.study_mode = nil if new_val != @wizard.course_id
+        @wizard.course_id = new_val
+      end
```

Comment:
Can we use a more descriptive variable name? As we are checking that it's not identical to the wizard's course, I think course_id is sufficient without the `new` prefix.
```suggestion
      def reset_study_mode_if_course_has_changed_and_update_course
        course_id = course_params['course_id']
        @wizard.study_mode = nil if course_id != @wizard.course_id
        @wizard.course_id = course_id
      end
```